### PR TITLE
Add a define_platforms configuration field

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: lib/src/backend/test_platform.dart [hunk 1]
```diff
@@ -109,6 +140,14 @@ class TestPlatform {
   Object serialize() {
     if (builtIn.contains(this)) return identifier;
 
+    if (parent != null) {
```
Comment (grouma) [inline lib/src/backend/test_platform.dart]: [optional] create the map with the fields name and identifier then conditionally append the parent field before returning.

File: doc/configuration.md [hunk 2]
```diff
@@ -567,6 +570,91 @@ when running on a particular operating system, use [`on_os`](#on_os) instead.
...
+### Browser Settings
+
+All built-in browser platforms provide the same settings that can be set using
+[`define_platforms`](#define_platforms), which control how the browser
+executable is invoked.
+
+#### `executable`
+
+The `executable` field tells the test runner where to look for the executable to
+use to start the browser. It has three sub-keys, one for each supported operating
```
Comment (grouma) [inline doc/configuration.md]: Should we mention that the sub-keys don't all have to be defined?

Comment (nex3) [reply doc/configuration.md]: Done.

File: lib/src/backend/test_platform.dart (hunk 1 above)
Comment (nex3) [reply lib/src/backend/test_platform.dart]: I think that would make it less clear that the child map isn't the same structure as the base-level map.

**Pull request merged**